src/javascript/DesignSystem/TreeView/TreeView.jsx

```diff
@@ -42,31 +42,44 @@ const style = theme => ({
     }
 });
 
-const TreeViewCmp = ({tree, onNodeClick, classes}) => {
+const TreeViewCmp = ({tree, onNodeClick, onNodeDoubleClick, classes}) => {
     // By default everything is closed
     const [openedNodes, setOpenedNode] = useState([]);
 
     function generateLevelJSX(level, deep) {
         return level.map(node => {
             const nodeIsOpen = Boolean(openedNodes.includes(node));
 
-            const handleNodeClick = e => {
+            const toogleNode = () => {
                 if (nodeIsOpen) {
                     setOpenedNode(openedNodes.filter(n => n !== node));
                 } else {
                     setOpenedNode([...openedNodes, node]);
                 }
+            };
 
+            const handleNodeClick = e => {
+                toogleNode();
                 onNodeClick(node, e);
             };
 
+            const handleNodeDoubleClick = e => {
+                toogleNode();
+                onNodeDoubleClick(node, e);
+            };
+
             if (node.childs && node.childs.length !== 0) {
                 const Arrow = nodeIsOpen || node.opened ? ArrowDropDown : ArrowRight;
                 const Childs = nodeIsOpen || node.opened ? generateLevelJSX(node.childs, deep + 1) : <></>;
 
                 return (
                     <Fragment key={level.id + node.id}>
-                        <button type="button" className={classes.nodeWithChild} onClick={handleNodeClick}>
+                        <button
```

Comment:
Why we don't use Button component ?

Reply:
Because it doesn't look like our Button Component

Reply:
okay hhh